Aesop/Search/ExpandSafePrefix.lean

```diff
@@ -18,7 +18,7 @@ private def Goal.isSafeExpanded (g : Goal) : BaseIO Bool :=
 mutual
   private partial def expandSafePrefixGoal (gref : GoalRef) :
       SearchM Q Unit :=
-    withIncRecDepth do
+    withIncRecDepth do withIncRecDepth do
```

Comment:
Is this double `withIncRecDepth` intentional?

Reply:
Yes, AFAICT the `Safe.lean` test checks that `expandSafePrefix` works on nonterminating rule sets without crashing.  This change makes the function detect the deep recursion before running out of stack space.

(This is probably not the what the test is supposed to do.  But it's what happens right now on master; `expandSafePrefix` produces subgoals with tags like `a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.....`.)

Reply:
I can't reproduce this; for me `Safe.lean` is fine with one `withIncRecDepth` as well. Is it possible that we are working with different max rec depths or different max stack depths?

Anyway, I'll merge the PR for now so that mathlib can switch back to this branch. Thanks again!